x-pack/legacy/plugins/integrations_manager/server/packages/fields.test.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { readFileSync } from 'fs';
+import { safeLoad } from 'js-yaml';
+
+// This should become a copy of https://github.com/elastic/beats/blob/d9a4c9c240a9820fab15002592e5bb6db318543b/libbeat/mapping/field.go#L39
+export interface Field {
+  keyword: string;
+  type: string;
+  required: boolean;
+  description: string;
+  fields: Field[];
+}
+
+test('test reading fields.yml', () => {
+  const yaml = readFileSync(__dirname + '/tests/fields.yml');
```

Comment:
[`path.join`](https://nodejs.org/docs/latest/api/path.html#path_path_join_paths) is preferred over string manipulation

```suggestion
  const yaml = readFileSync(path.join(__dirname, '/tests/fields.yml'));
```

It's got some intelligence around dealing with paths like multiple directory separators, etc

```javascript
const dir = '/some/dir/'
const pathToFile = '/path/to/file.ext'
console.log(path.join(dir, pathToFile)) # '/some/dir/path/to/file.ext'
console.log(dir + pathToFile)           # '/some/dir//path/to/file.ext'
```

Reply:
same in golang, I was just lazy ;-)